libsolidity/codegen/LValue.cpp

```diff
@@ -357,38 +357,45 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 				"Struct assignment with conversion."
 			);
 			solAssert(!structType.containsNestedMapping(), "");
-			solAssert(sourceType.location() != DataLocation::CallData, "Structs in calldata not supported.");
-			for (auto const& member: structType.members(nullptr))
+			if (sourceType.location() == DataLocation::CallData)
 			{
-				// assign each member that can live outside of storage
-				TypePointer const& memberType = member.type;
-				solAssert(memberType->nameable(), "");
-				TypePointer sourceMemberType = sourceType.memberType(member.name);
-				if (sourceType.location() == DataLocation::Storage)
+				m_context << Instruction::DUP2 << Instruction::DUP2;
```

Comment:
Please add assertion about the size on the stack of the source type.

Reply:
Added both for source and target type